docs/src/model_creation/events_and_callbacks.md

```diff
@@ -0,0 +1,497 @@
+
+Here, we create a birth-death process where we add `2.0` additional units of `X` through a preset-time callback. At the end of the `affect!` function, we also call `reset_aggregated_jumps!` on the integrator:
+```@example events_considerations
+using DiffEqCallbacks # hide
+using JumpProcesses
+rn = @reaction_network begin
+  (p,d), 0 <--> X
+end
+
+u0 = [:X => 1]
+tspan = (0.0, 10.0)
+ps = [:p => 1.0, :d => 0.2]
+dprob = DiscreteProblem(bd_model, u0, tspan, ps)
+jprob = JumpProblem(bd_model, dprob, Direct())
+
+condition = [3.0, 7.0]
+function affect!(integrator)
+    integrator[:X] += 2.0
+    reset_aggregated_jumps!(integrator)
+end
+cb = PresetTimeCallback(condition, affect!)
+
+sol = solve(jprob, SSAStepper(); callback = cb)
+plot(sol)
+```
+
+### [Events occurring at specific time points](@id events_additional_considerations_tstops)
+Typically, events that occur at designated time points should be implemented through [preset-time events](@ref events_model_discrete_presettime) (or [preset-time callbacks](@ref events_model_callbacks_presettime)). However, sometimes one might want to use a a discrete event (or callback) instead. This is a potential problem, as discrete events (and callbacks) are only checked at the end of each time step. If you have defined a discrete event which is triggered when `t == 5.0`, but the simulation time stepper stops at $t = ... 4.9007, 5.0214, 5.1094, ...$, the condition won't hold, and is never triggered. Here, you have to explicitly instruct the simulation to make a stop at the specific time using the `tstops` argument.
+
+In this example we wish to introduce an event which is triggered at time `5.0`, but only if `X > 2.0`. First, we create an `ODEProblem`:
+```@example events_considerations
+bd_model = @reaction_network begin
+  @discrete_events begin
+    (X > 2.0) & (t == 5.0) => [X ~ 2.0]
+  end
+  (p,d), 0 <--> X
+end
+
+u0 = [:X => 0.1]
+tspan = (0.0, 10.0)
+ps = [:p => 2.0, :d => 0.2]
+
+oprob = ODEProblem(bd_model, u0, tspan, ps)
+nothing # hide
+```
+If we now simulate the model normally, the event is not triggered:
+```@example events_considerations
+sol = solve(oprob)
+plot(sol)
+```
+However, if we instruct `solve` to specifically stop at time `5.0`, it *is* triggered:
+```@example events_considerations
+sol = solve(oprob; tstops = [5.0])
+plot(sol)
+```
+Here, `tstops` takes a vector with all the time points we wish to stop at (here we only designate `5.0`).
+
+Since continuous events (and callbacks) actively find the exact time point when they are triggered, supplying `tstops` is not required for these.
+
+## [Programmatic creation of models containing events](@ref events_programmatic)
```

Comment:
Right now, I start with going through how events work with the DSL. Then, at the end I have a separate section describing the details one needs to change when using a programmatic workflow (here and most other cases it is straightforward, except for the base syntax the two approaches are really similar).